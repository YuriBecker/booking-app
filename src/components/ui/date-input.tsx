import { format } from "date-fns";
import { Calendar as CalendarIcon } from "lucide-react";
import { Button } from "@/components/ui/button";
import { Calendar } from "@/components/ui/calendar";
import {
  Popover,
  PopoverContent,
  PopoverTrigger,
} from "@/components/ui/popover";
import { cn } from "@/utils/tailwind";
import { SelectSingleEventHandler } from "react-day-picker";

interface Props {
  onChange: SelectSingleEventHandler;
  value?: Date;
  id?: string;
  placeholder: string;
}

export function DateField({ placeholder, id, value, onChange }: Props) {
  return (
    <Popover>
      <PopoverTrigger asChild>
        <Button
          variant={"outline"}
          className={cn(
            "w-full justify-start text-left font-normal",
            !value && "text-muted-foreground"
          )}
          id={id}
        >
          <CalendarIcon className="mr-2 h-4 w-4" />
          {value ? format(value, "PPP") : <span>{placeholder}</span>}
        </Button>
      </PopoverTrigger>
      <PopoverContent className="w-auto p-0">
        <Calendar
          mode="single"
          selected={value}
          onSelect={onChange}
          initialFocus
          disabled={[
            {
              before: new Date(),
            },
          ]}
        />
      </PopoverContent>
    </Popover>
  );
}
