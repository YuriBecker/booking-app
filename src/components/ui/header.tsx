import routerPaths from "@/router/paths";
import clsx from "clsx";
import { useEffect, useRef, useState } from "react";
import { Button } from "./button";
import { House } from "lucide-react";

const Header = () => {
  const headerRef = useRef<HTMLHeadingElement>(null);

  const [isScrolled, setIsScrolled] = useState(false);

  useEffect(() => {
    const handleScroll = () => {
      if (window.scrollY > 30) {
        setIsScrolled(true);
      } else {
        setIsScrolled(false);
      }
    };

    window.addEventListener("scroll", handleScroll);

    return () => {
      window.removeEventListener("scroll", handleScroll);
    };
  }, []);

  return (
    <header
      className={clsx(
        "flex items-center h-[100px] bg-secondary sticky top-[-30px] z-50",
        isScrolled && "shadow-lg"
      )}
      ref={headerRef}
    >
      <div className="container mx-auto  h-[70px] sticky top-0 flex items-center justify-between gap-8">
        <a href={routerPaths.home}>
          <img
            className={clsx(
              "w-full h-10 md:h-14 transform transition-all duration-200 hover:scale-105",
              isScrolled ? "h-10" : "h-12"
            )}
            src="/logo.svg"
            alt="hostfully"
          />
        </a>

        <div className="flex items-center">
          <Button variant="default">
            <House className="mr-2" /> My Bookings
          </Button>
        </div>
      </div>
    </header>
  );
};

export default Header;
