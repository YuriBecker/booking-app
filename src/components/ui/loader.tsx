import Lottie from "react-lottie";
import animationData from "public/loader-animation.json";

type Props = {
  size?: number;
  className?: string;
  fixed?: boolean;
};

const defaultOptions = {
  loop: true,
  autoplay: true,
  animationData,
  rendererSettings: {
    preserveAspectRatio: "xMidYMid slice",
  },
};

export const Loader = ({ size = 120, fixed }: Props) => {
  const containerClass = fixed
    ? "fixed inset-0 flex items-center justify-center"
    : "";

  return (
    <div className={containerClass}>
      <Lottie
        options={defaultOptions}
        height={size}
        width={size}
        isClickToPauseDisabled
      />
    </div>
  );
};
