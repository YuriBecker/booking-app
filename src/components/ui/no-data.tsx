import Lottie from "react-lottie";
import animationData from "public/no-data-animation.json";
import { cn } from "@/utils/tailwind";
import { Button } from "./button";

type Props = {
  size?: number;
  className?: string;
  onClick?: () => void;
  buttonLabel?: string;
};

const defaultOptions = {
  loop: false,
  autoplay: true,
  animationData,
  rendererSettings: {
    preserveAspectRatio: "xMidYMid slice",
  },
};

const NoData = ({ size, className, onClick, buttonLabel }: Props) => {
  return (
    <div
      className={cn(
        "mx-auto flex flex-col items-center mt-10 max-w-full w-[450px]",
        className
      )}
    >
      <h1 className="text-3xl font-bold text-secondary text-center">
        Sorry, we couldn't find any available property
      </h1>

      <div className="pointer-events-none">
        <Lottie
          options={defaultOptions}
          height={size}
          width={size}
          isClickToPauseDisabled
        />
      </div>

      <Button variant="secondary" onClick={onClick} className="text-xl">
        {buttonLabel}
      </Button>
    </div>
  );
};

export default NoData;
