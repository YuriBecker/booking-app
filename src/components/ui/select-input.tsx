import * as React from "react";
import { cn } from "@/utils/tailwind";
import { Check, ChevronsUpDown, Search } from "lucide-react";
import { Button } from "@/components/ui/button";
import {
  Command,
  CommandEmpty,
  CommandGroup,
  CommandInput,
  CommandItem,
  CommandList,
} from "@/components/ui/command";
import {
  Popover,
  PopoverContent,
  PopoverTrigger,
} from "@/components/ui/popover";

type Option = {
  value: string;
  label: string;
};

interface Props
  extends Pick<React.InputHTMLAttributes<HTMLSelectElement>, "id" | "value"> {
  onChange: (value: string) => void;
  options: Option[];
  placeholder?: string;
}

export function SelectField({
  id,
  value,
  onChange,
  options,
  placeholder,
}: Props) {
  const [open, setOpen] = React.useState(false);

  return (
    <Popover open={open} onOpenChange={setOpen}>
      <PopoverTrigger asChild>
        <Button
          variant="outline"
          role="combobox"
          aria-expanded={open}
          className="w-full justify-between"
          id={id}
        >
          <div className="flex flex-row items-center">
            <Search width={16} className="mr-2" />
            {value
              ? options.find((city) => city.value === value)?.label
              : placeholder}
          </div>
          <ChevronsUpDown className="ml-2 h-4 w-4 shrink-0 opacity-50" />
        </Button>
      </PopoverTrigger>
      <PopoverContent className="w-[200px] p-0">
        <Command>
          <CommandInput placeholder="Search city..." />
          <CommandEmpty>No city found.</CommandEmpty>
          <CommandGroup>
            <CommandList>
              {options.map((city) => (
                <CommandItem
                  key={city.value}
                  value={city.value}
                  onSelect={(currentValue) => {
                    if (onChange)
                      onChange(currentValue === value ? "" : currentValue);

                    setOpen(false);
                  }}
                >
                  <Check
                    className={cn(
                      "mr-2 h-4 w-4",
                      value === city.value ? "opacity-100" : "opacity-0"
                    )}
                  />
                  {city.label}
                </CommandItem>
              ))}
            </CommandList>
          </CommandGroup>
        </Command>
      </PopoverContent>
    </Popover>
  );
}
