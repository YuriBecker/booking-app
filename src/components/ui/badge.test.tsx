import { render } from "@testing-library/react";
import { Badge } from "@/components/ui/badge";

test("Badge component renders correctly with default variant", () => {
  const { asFragment } = render(<Badge>Hello</Badge>);
  expect(asFragment()).toMatchSnapshot();
});

test("Badge component renders correctly with secondary variant", () => {
  const { asFragment } = render(<Badge variant="secondary">Hello</Badge>);
  expect(asFragment()).toMatchSnapshot();
});

test("Badge component renders correctly with destructive variant", () => {
  const { asFragment } = render(<Badge variant="destructive">Hello</Badge>);
  expect(asFragment()).toMatchSnapshot();
});

test("Badge component renders correctly with outline variant", () => {
  const { asFragment } = render(<Badge variant="outline">Hello</Badge>);
  expect(asFragment()).toMatchSnapshot();
});
