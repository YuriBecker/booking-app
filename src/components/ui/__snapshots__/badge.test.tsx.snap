// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge component > renders correctly with default variant 1`] = `
<DocumentFragment>
  <div
    class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-primary text-primary-foreground hover:bg-primary/80"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`Badge component > renders correctly with destructive variant 1`] = `
<DocumentFragment>
  <div
    class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-destructive text-destructive-foreground hover:bg-destructive/80"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`Badge component > renders correctly with outline variant 1`] = `
<DocumentFragment>
  <div
    class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-foreground"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`Badge component > renders correctly with secondary variant 1`] = `
<DocumentFragment>
  <div
    class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80"
  >
    Hello
  </div>
</DocumentFragment>
`;
